sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -1115,6 +1115,7 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
         Row(null, null, null),
         Row(null, null, null),
         Row(null, null, null),
+        Row(null, null, null),
```

Comment:
so for json data source, previous behavior is, we would skip the row even it's in PERMISSIVE mode. Shall we clearly mention it in the migration guide?

Reply:
> so for json data source, previous behavior is, we would skip the row even it's in PERMISSIVE mode.

Yes, we skipped such rows if `Jackson` parser wasn't able to find any root tokens. So, not only empty strings and gaps got into the category.

> Shall we clearly mention it in the migration guide?

Sure.